### ISSUE TITLE
How to use the generated embedding? 

### ISSUE THREAD

**Issue by sleebapaul** [luuil/tensorflow-audio-classification#1]
I've generated the embedding using VGGish model for .wav files (16 kHz sampling frequency and 2 seconds long). The shape is embedding is `(2x128)`. 

I would like to add another lite classifier to be trained in these embeddings to classify the labels. Have you tried this approach? If so, can you describe the architecture of the lite network? Are CNNs involved in the network? 

I'm interested in using MobileNet / SqueezeNet for on-premise prediction. But I'm worried about the data input as I can't treat a `(2x128)`  as an image.

**Comment by luuil** [comment]
Sorry for late response.

Actually, you can easily find the lite network architecture in my code if you have read the whole project. It lives in `/audio/audio_model.py`, that's what you ask for. 

The architecture is very simple, That is, `two fully connected layers and one softwaremax layer`. In addition, it does not involves CNNs. Because, for a such simple classification tasks, I personally think embedding is so compact that it does not require a complex networks, such as CNNs.

**Comment by sleebapaul** [comment]
Cool. My bad that I couldn't check it. I've set up a simple network of mine with a CNN layer, a Max pool, and a fully connected layer. It works with ~90% accuracy. Thanks for the reply. Out of context, I would like to ask you a general information. 

Imagine I have a dataset of 1000 categories. I want to build a lite classifier from it, with a class of 3. 
Say, 
`Dogs`, `Cats`, and `Others`.  
Obviously, in this approach, there will be a confusion. `Others` category will be having much more examples than `Dogs` or `Cats` and it is going to be diverse. To make things worse, we could expect many noises which are not in `Others` in the training set when it comes to testing. 

What do you think about confronting this situation? How can we pull this off?  (I'm asking this in context of Audio Dataset.)

**Comment by luuil** [comment]
So, you did it yourself, it's great!

As for data imbalance problem, especially audio datasets, I'm not very experienced. But I've found some hints on the Internet that you can try it out: [how-to-balance-my-dataset](https://stats.stackexchange.com/questions/152823/how-to-balance-my-dataset).

**Comment by JingpengSun** [comment]
> Cool. My bad that I couldn't check it. I've set up a simple network of mine with a CNN layer, a Max pool, and a fully connected layer. It works with ~90% accuracy. Thanks for the reply anyway. Out of context, I would like to ask you a general information.
> 
> Imagine I have a dataset of 1000 categories. I want to build a lite classifier from it, with a class of 3.
> Say,
> `Dogs`, `Cats`, and `Others`.
> Obviously, in this approach, there will be a confusion. `Others` category will be having much more examples than `Dogs` or `Cats` and it is going to be diverse. To make things worse, we could expect many noises which are not in `Others` in the training set when it comes to testing.
> 
> What do you think about confronting this situation? How can we pull this off? (I'm asking this in context of Audio Dataset.)

Hi, I'm very interested in your model with CNN.  I did not find it in your github homepage. Do you mind telling me where I can find it? Thanks

**Comment by sleebapaul** [comment]
I'm afraid I can't help you with the code. It is a private repository as of now.

**Comment by JingpengSun** [comment]
> I'm afraid I can't help you with the code. It is a private repository as of now.

Oh, I see. Thank you all the same for your reply.
